pkg/scheduler/controller/shoot/scheduler_control.go

```diff
@@ -175,61 +176,98 @@ func determineSeed(shoot *gardencorev1beta1.Shoot, seedLister gardencorelisters.
 		return nil, err
 	}
 
-	seedsMatchingCloudProfileSelector, err := filterSeedsMatchingSeedSelector(seedList, cloudProfile.Spec.SeedSelector, "CloudProfile")
-	if err != nil {
-		return nil, err
+	filteredSeeds, err := filterUsableSeeds(seedList)
+	if err == nil {
+		filteredSeeds, err = filterSeedsMatchingSeedSelector(filteredSeeds, cloudProfile.Spec.SeedSelector, "CloudProfile")
 	}
-	seedsMatchingShootSelector, err := filterSeedsMatchingSeedSelector(seedsMatchingCloudProfileSelector, shoot.Spec.SeedSelector, "Shoot")
-	if err != nil {
-		return nil, err
+	if err == nil {
+		filteredSeeds, err = filterSeedsMatchingSeedSelector(filteredSeeds, shoot.Spec.SeedSelector, "Shoot")
 	}
-
-	candidates, err := getCandidates(shoot, seedsMatchingShootSelector, strategy)
-	if err != nil {
-		return nil, err
+	if err == nil {
+		filteredSeeds, err = filterSeedsMatchingProviders(cloudProfile, shoot, filteredSeeds)
+	}
+	if err == nil {
+		filteredSeeds, err = filterCandidates(shoot, filteredSeeds)
 	}
+	if err == nil {
+		filteredSeeds, err = applyStrategy(shoot, filteredSeeds, strategy)
+	}
+	if err == nil {
+		return getSeedWithLeastShootsDeployed(filteredSeeds, shootList)
+	}
+	return nil, err
```

Comment:
TBH, I get the motivation for this, but I would still prefer the standard style where the error is handled first. This is what we use all over the places.

Reply:
Just trying ....